addons/purchase_stock/models/purchase_order.py

```diff
@@ -124,27 +124,25 @@ def _get_action_add_from_catalog_extra_context(self):
             'vendor_suggest_days': self.partner_id.suggest_days,
             'vendor_suggest_based_on': self.partner_id.suggest_based_on,
             'vendor_suggest_percent': self.partner_id.suggest_percent,
-            'po_state': self.state,
         }
 
-    def action_purchase_order_suggest(self):
+    def action_purchase_order_suggest(self, search_domain, section_id):
         """ Adds suggested products to PO, removing products with no suggested_qty, and
         collapsing existing po_lines into at most 1 orderline. Saves suggestion params
         (eg. number_of_days) to partner table. """
         self.ensure_one()
         ctx = self.env.context
-        domain = [('type', '=', 'consu')]
-        if ctx.get("domain"):
-            domain = fields.Domain.AND([domain, ctx.get("domain")])
-
-        products = self.env['product.product'].search(domain)
-
         self.partner_id.write({
             'suggest_days': ctx.get('suggest_days'),
             'suggest_based_on': ctx.get('suggest_based_on'),
             'suggest_percent': ctx.get('suggest_percent'),
         })
 
+        domain = [('type', '=', 'consu')]
+        if search_domain:
+            domain = fields.Domain.AND([domain, search_domain])
+        products = self.env['product.product'].search(domain)
+
```

Comment:
Why moved down ?

Reply:
@jeanschoenlaub Ok I understand, but to avoid useless change I reverted this part since we try to avoid to change/clean code in stable.
That said, I'm not sure to understand the two actions RPC thing ? You can create a separate method and call it from this method if you want, it won't cause any additional RPC